main/src/main/java/org/mobilitydata/gtfsvalidator/validator/ExpiredCalendarValidator.java

```diff
@@ -54,12 +53,36 @@ public void validate(NoticeContainer noticeContainer) {
     final Map<String, SortedSet<LocalDate>> servicePeriodMap =
         CalendarUtil.servicePeriodToServiceDatesMap(
             CalendarUtil.buildServicePeriodMap(calendarTable, calendarDateTable));
+    boolean isCalendarTableEmpty = calendarTable.getEntities().isEmpty();
+    List<ExpiredCalendarNotice> expiredCalendarDatesNotices = new ArrayList<>();
+    boolean allCalendarAreExpired = true;
     for (var serviceId : servicePeriodMap.keySet()) {
       SortedSet<LocalDate> serviceDates = servicePeriodMap.get(serviceId);
-      if (!serviceDates.isEmpty() && serviceDates.last().isBefore(dateForValidation.getDate())) {
-        int csvRowNumber = calendarTable.byServiceId(serviceId).get().csvRowNumber();
-        noticeContainer.addValidationNotice(new ExpiredCalendarNotice(csvRowNumber, serviceId));
+      if (serviceDates.isEmpty()) {
+        continue;
       }
+      if (serviceDates.last().isBefore(dateForValidation.getDate())) {
+        if (calendarTable.byServiceId(serviceId).isPresent()) {
```

Comment:
Nitpicking here. In the case you know the calendarTable is empty (isCalendarTableEmpty == true), why bother doing a lookup on it?
```suggestion
        if (!isCalendarTableEmpty && calendarTable.byServiceId(serviceId).isPresent()) {
```

Reply:
That's true. However, reducing the logic in the if condition helps readability. In either case, empty or not, the _byService(serviceId)_ method will behave the same.